src/chm/documents/customactions/osinfo.html.md

```diff
@@ -549,3 +549,17 @@ WixUtilExtension will automatically schedule the custom actions as needed after
 <td>Set to 1 if the target machine has composition enabled. This property is only set on machines running Windows Vista or higher.</td>
 </tr>
 </table>
+
+<h2>WixQueryNativeMachine properties</h2>
+
+<table>
+<tr>
+<td valign="top">
+<p>WIX_NATIVE_MACHINE</p>
+</td>
+
+<td>
+<p>Set to an [Image File Machine value](https://docs.microsoft.com/en-us/windows/win32/sysinfo/image-file-machine-constants) representing the native architecture of the machine. This property is only set on Windows 10, version 1511 (TH2) and higher.</p>
```

Comment:
This is the only functional problem I see with the PR - the documentation says that the MSI property and Burn variable is only set on Windows 10, version 1511 and higher but they are always set. I vote for the implementation to match the documentation and not set the property or variable if the API isn't available.

I can see you were trying to follow the pattern from `WIX_WDDM_DRIVER_PRESENT` and `WIX_DWM_COMPOSITION_ENABLED`. If you look at [DetectWDDMDriver](https://github.com/wixtoolset/wix3/blob/58abd6993afba08b39e37b0e76b1790161df9231/src/ext/ca/wixca/dll/OsInfo.cpp#L382) and [DetectIsCompositionEnabled](https://github.com/wixtoolset/wix3/blob/58abd6993afba08b39e37b0e76b1790161df9231/src/ext/ca/wixca/dll/OsInfo.cpp#L420), if the corresponding API could not be dynamically loaded then it returns successfully without setting the property.

It also looks like you implemented `ProcNativeMachine` following the pattern of `ProcWow64`. But this conflicts with the pattern above because the caller can't tell whether the API was available or not. `ProcWow64` reports success and returns `FALSE` if the API wasn't available, which makes sense because there's no way it could be running on WOW in that case. Maybe `ProcNativeMachine` should return `E_NOTIMPL` or maybe there's no need to tell the difference between the API not being available and `IsWow64Process2` returning `IMAGE_FILE_MACHINE_UNKNOWN`, I don't know.

Reply:
Good catch.  You are right that today it will set IMAGE_FILE_MACHINE_UNKNOWN (0) rather than not setting the value when the API is not supported.

> maybe there's no need to tell the difference between the API not being available and IsWow64Process2 returning IMAGE_FILE_MACHINE_UNKNOWN, I don't know.

In our case we don't need to distinguish, but I could see someone wanting to know this.

I am fine with changing it to not set the property at all and make it consistent with the docs.